x-pack/test/functional_endpoint/apps/endpoint/resolver.ts

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function({ getPageObjects, getService }: FtrProviderContext) {
+  const pageObjects = getPageObjects(['common', 'timePicker', 'endpointAlerts']);
+  const testSubjects = getService('testSubjects');
+  const esArchiver = getService('esArchiver');
+  const retry = getService('retry');
+  const browser = getService('browser');
+
+  describe('Endpoint Alert Resolver', function() {
+    this.tags(['ciGroup7']);
+    before(async () => {
+      const fromTime = 'Sep 22, 2019 @ 20:31:44.000';
+      const toTime = 'Now';
+      await esArchiver.load('endpoint/resolver_tree/api_feature');
+      await pageObjects.common.navigateToUrlWithBrowserHistory('endpoint', '/alerts');
+      await pageObjects.common.sleep(4000);
+      await pageObjects.timePicker.setAbsoluteRange(fromTime, toTime);
+      await testSubjects.existOrFail('alertListPage');
+      await (await testSubjects.find('alertTypeCellLink')).click();
+      await testSubjects.existOrFail('alertDetailFlyout');
+      await (await testSubjects.find('overviewResolverTab')).click();
+      await pageObjects.common.sleep(2000);
```

Comment:
care to put a comment in as to why its needed?  Also, here in the PR you could chat why / how it was assessed and if we really think we can't do a better wait state?

Reply:
Why do you not like the sleep
I can replace the first sleep with:
await retry.try(async function() {
await testSubjects.existOrFail('superDatePickerShowDatesButton');
});

But the problem is it will run non stop until it finds the "superDatePickerShowDatesButton"

Reply:
The second one does not work this way.

Reply:
I fixed I replaced with Wait